### PR TITLE
add osxarm

### PR DESCRIPTION
**Pull request by ngam**
- MNT: Re-rendered with conda-build 3.21.7, conda-smithy 3.16.1, and conda-forge-pinning 2022.01.03.16.37.21
- add osx

### CODE REVIEW

**Comment by ngam** [comment]
@bernt-matthias, thanks for the good work here, could merge this if it passes? I simply added support for osx arm65, following this: https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/#how-to-add-a-osx-arm64-build-to-a-feedstock

edit: see meta.yaml, I added this to make it pass 

```yaml
    # tcsh does NOT recognize arm64 and expect arm
    # this sed call just edits the config.sub file to
    # replace all mentions of arm to arm64 and thus faking it to pass
    # however, this also leads to a version: unknown-apple-darwin in test
    # but, we don't test that... we should probably file an issue upstream...
    # issue upstream: https://github.com/tcsh-org/tcsh/issues/46
    - mv config.sub config.sub_old; sed 's/arm/arm64/g' config.sub_old > config.sub  # [osx and arm64]

```

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,6 +13,13 @@ build:
   number: 0
   skip: true  # [win]
   script:
+    # tcsh does NOT recognize arm64 and expect arm
+    # this sed call just edits the config.sub file to
+    # replace all mentions of arm to arm64 and thus faking it to pass
+    # however, this also leads to a version: unknown-apple-darwin in test
+    # but, we don't test that... we should probably file an issue upstream...
+    # issue upstream: https://github.com/tcsh-org/tcsh/issues/46
+    - mv config.sub config.sub_old; sed 's/arm/arm64/g' config.sub_old > config.sub  # [osx and arm64]
```
Comment (bernt-matthias) [inline recipe/meta.yaml]:
```suggestion
    - sed -i -e 's/arm/arm64/g' config.sub
```

should work as well

Comment (ngam) [reply recipe/meta.yaml]: Thanks for the tip --- new to sed :)

**Comment by ngam** [comment]
@bernt-matthias not sure if you noticed but your merge from yesterday failed for ppc64le (travis ci wasn't working properly). You can see this if you go to the master ---- there will be a red x next to the commit: 

bernt-matthias Merge pull request #8 from regro-cf-autotick-bot/6_23_02_h5770a0 … X 27040a9 

So I changed ppc64le to azure instead of travis. Should be more reliable.

**Comment by bernt-matthias** [comment]
> @bernt-matthias not sure if you noticed but your merge from yesterday failed for ppc64le (travis ci wasn't working properly). You can see this if you go to the master ---- there will be a red x next to the commit:
> 
> bernt-matthias Merge pull request #8 from regro-cf-autotick-bot/6_23_02_h5770a0 … X [27040a9](https://github.com/conda-forge/tcsh-feedstock/commit/27040a9cad5826d1c092f3ad3fa97e470de1b31f)
> 
> So I changed ppc64le to azure instead of travis. Should be more reliable.

Jep, I noticed but had no time to check so far. I restarted the failing job. Lets see if this helps.

**Comment by ngam** [comment]
> Lets see if this helps

Looks like something is wrong with travis and I noticed some other maintainers switching to azure.

Anyway, feel free to merge this if you're satisfied, and let me know if you need help. There should be a fix for arm64-apple vis-a-vis arm-apple coming up in the next release https://github.com/tcsh-org/tcsh/issues/46

**Pull request merged**